### PR TITLE
Introduce Claim Revision with sequence diagrams

### PR DESCRIPTION
**Pull request by pepermao**
- [x] Claim creation
- [x] Claim update

### CODE REVIEW

File: server/claim/claim.service.ts [hunk 1]
```diff
@@ -38,7 +38,6 @@ export class ClaimService {
             .sort({ _id: order })
             .lean();
         return Promise.all(
-            /** Get lastest revision for each claim */
             claims.map(async (claim) => {
```
Comment (thesocialdev) [inline server/claim/claim.service.ts:41]: Here's a place where we forgot to update with the new `this._getClaim` functionality.

File: server/personality/personality.service.ts [hunk 2]
```diff
@@ -95,7 +93,7 @@ export class PersonalityService {
         personality.claims = await Promise.all(personality.claims.map((claim) => {
             return {
                 ...claim,
-                ...claim.revisions[0]
+                ...claim.lastestRevision
```
Comment (thesocialdev) [inline server/personality/personality.service.ts:96]: we should clean-up the query above, although this is working, we don't need to populate revisions the way we did before

**Review by thesocialdev — CHANGES_REQUESTED** [review]

File: server/claim/claim.service.ts (hunk 1 above)
Comment (pepermao) [reply server/claim/claim.service.ts:41]: I didn't understand

File: server/claim/claim.service.ts [hunk 3]
```diff
@@ -46,80 +54,78 @@ export class ClaimService {
...
-
-        return this.postProcess(claim.toObject());
+        return this._getClaim({_id: claimId})
     }
 
     async getByPersonalityIdAndClaimSlug(personalityId, claimSlug) {
+        return this._getClaim({personality: personalityId, slug: claimSlug})
+    }
+
+    private async _getClaim(match: ClaimMatchParameters, postprocess = true) {
```
Comment (thesocialdev) [inline server/claim/claim.service.ts:101]: @pepermao I meant that we have created this new function that retrieves Claim with a revision, and execute the postProcess function.

In the comment above we should use _getClaim and refactor the functionality.

Comment (thesocialdev) [reply server/claim/claim.service.ts:101]: But let's do this in a another PR because it's going to be a little tricky.

**Review by thesocialdev — APPROVED** [review]

**Pull request merged**